plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/PlainTextKafkaProducerFactory.java

```diff
@@ -60,6 +62,7 @@ public Properties configure(ConnectorSession session)
         properties.setProperty(ACKS_CONFIG, "all");
         properties.setProperty(LINGER_MS_CONFIG, Long.toString(5));
         properties.setProperty(SECURITY_PROTOCOL_CONFIG, securityProtocol.name());
+        properties.putAll(fileProperties);
```

Comment:
file properties should be loaded first

Reply:
The reason I placed it at the end is to allow users to override some configs from Trino. e.g. use the property `SECURITY_PROTOCOL_CONFIG=SASL_SSL`.

Reply:
This should be the other way round. We always treat config resource files as a base which is update. If user wants to use config from resource files then they should use defaults in kafka which should not update anything.

Reply:
I understand that you don't want to being overwritten by `SECURITY_PROTOCOL_CONFIG`. What kafka would use if we would skip setting this property at all?

Reply:
Default is `PLAINTEXT` from doc https://kafka.apache.org/documentation/#producerconfigs_security.protocol .

Reply:
In this case, do you agree if I load the file properties first and then remove `properties.setProperty(SECURITY_PROTOCOL_CONFIG, securityProtocol.name()); ` ?

Reply:
How about we would make `DefaultTextKafkaConsumerFactory` that would be injected into `PlainTextKafkaConsumerFactory` as delegate (like we do in `SslKafkaConsumerFactory`). The `DefaultTextKafkaConsumerFactory` would not set this property and the `DefaultTextKafkaConsumerFactory` should be default for Kafka. This way it won't override anything, but only load properties from file.

It would be awkward to explicitly require `PLAINTEXT` in `kafka.properties` that would be overwritten later to something else.

Reply:
Yes great idea! I think we will need to remove default `PLAINTEXT` [here](https://github.com/trinodb/trino/blob/master/plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaSecurityConfig.java#L30) to make this work.

To make sure we are on the same page, if `kafka.security-protocol` is omitted in `kafka.properties`, then we will use `DefaultTextKafkaXFactory` which loads from properties file correct?

Reply:
Yes.

Reply:
Would the default factories ever be used without the plain text ones? If not then it seems like it might be easier to add the external properties in the plain text factories and remove the default factories entirely. Correct me if I've misunderstood something

Reply:
Yes the default can be used to augment options that are not managed by Trino explicitly.

Reply:
The idea is to allow Trino users to provide different security protocols  that are not yet directly managed by Trino (e.g. `SASL_PLAINTEXT`, `SASL_SSL`).